packages/react-components/src/components/ControlBarButton.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { useCallback } from 'react';
+import { DefaultButton, IButtonProps, IRenderFunction, Label, concatStyleSets, mergeStyles } from '@fluentui/react';
+import { controlButtonLabelStyles, controlButtonStyles } from './styles/ControlBar.styles';
+
+/**
+ * Strings of ControlBarButton that can be overridden
+ */
+export interface ControlBarButtonStrings {
+  /**
+   * Label of the button. This supersedes onLabel or offLabel if used.
+   */
+  label?: string;
+  /**
+   * Label of the button shown when the button is checked.
+   */
+  onLabel?: string;
+  /**
+   * Label of the button shown when the button is not checked.
+   */
+  offLabel?: string;
+}
+
+/**
+ * Props for ControlBarButton component
+ */
+export interface ControlBarButtonProps extends IButtonProps {
+  /**
+   * Whether the label is displayed or not.
+   *
+   * @defaultValue `false`
+   */
+  showLabel?: boolean;
+
+  /**
+   * Key to use for the Label component
+   */
+  labelKey?: string;
+
+  /**
+   * Optional strings to override in component.
+   */
+  strings?: ControlBarButtonStrings;
+
+  /**
+   * Icon to render when the button is checked.
+   */
+  onRenderOnIcon?: IRenderFunction<IButtonProps>;
+
+  /**
+   * Icon to render when the button is not checked.
+   */
+  onRenderOffIcon?: IRenderFunction<IButtonProps>;
+}
+
+/**
+ * Default button styled for the Control Bar. This can be used to create custom buttons that are styled the same as other buttons provided by the UI Library.
+ */
+export const ControlBarButton = (props: ControlBarButtonProps): JSX.Element => {
+  const {
+    showLabel = false,
+    labelKey,
+    styles,
+    onRenderText,
+    onRenderIcon,
+    onRenderOnIcon,
+    onRenderOffIcon,
+    strings
+  } = props;
+
+  const defaultRenderText = useCallback(
+    (props?: IButtonProps): JSX.Element => {
+      const labelText = props?.text ?? strings?.label ?? (props?.checked ? strings?.onLabel : strings?.offLabel);
+      return (
+        <Label key={labelKey} className={mergeStyles(controlButtonLabelStyles, props?.styles?.label)}>
+          {labelText}
+        </Label>
+      );
+    },
+    [labelKey, strings?.label, strings?.offLabel, strings?.onLabel]
```

Comment:
Why the difference of treatment between `props.styles.labels` vs `props.strings.label` etc?

Contrast with just a private module level functions that takes `props` as arguments. Or even better, a `DefaultRenderText` component with `props`.

Reply:
Same comment for `defaultRenderIcon` below.

Reply:
Do you mean why are some left out of this dependency array? If so - total oversight and I'll add them now, I actually thought the eslint rule for hooks was meant to catch this but it isn't being applied to useCallback...

Reply:
Actually props don't need in there because its a function taking in props (still odd there's no linting to make this decision). Did you mean why are some prefixed with `props` and others extracted in the props above? I have no good answer :D I will update to none are extracted in the props above I think

Reply:
Yeah, I meant the latter. Some are captured by the closure, while others are used from the `props` arg, and it isn't clear to me why there is that difference.

As I said, this really looks like a candidate for a private `DefaultRenderTest` component to me.

Reply:
Oh yah that's much cleaner now, nice suggestion